src/odemis/gui/cont/tabs.py

```diff
@@ -1840,7 +1841,21 @@ def display_new_data(self, filename, data, extend=False):
         all_streams = streams + self.tab_data_model.streams.value
 
         # Spectrum and AR streams are, for now, considered mutually exclusive
+
+        # all spectrum streams including spectrum, temporal spectrum and time correlator (chronograph)
         spec_streams = [s for s in all_streams if isinstance(s, acqstream.SpectrumStream)]
+        self.temporalspectrum_streams = []
+        self.spectrum_streams = []
+        self.timecorrelator_streams = []
```

Comment:
No, please, don't store the stream in variable. It's kind of handy, but now, the streams have another point of reference, so they will never stay in memory until the lists are reset (by calling this method again).
It's safer to just separate all the 3 types of streams every time. If you need to do it often, you could put the code in a method called like `_get_time_spectrum_streams()`, which returns the 3 lists.